starknet-rpc-test/add_deploy_account_transaction.rs

```diff
@@ -46,55 +48,58 @@ async fn fail_execution_step_with_no_storage_change(#[future] madara: MadaraClie
 
 #[rstest]
 #[tokio::test]
-async fn works_with_storage_change(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
-    let madara = madara.await;
-    let rpc = madara.get_starknet_client();
+async fn works_with_storage_change(madara: &ThreadSafeMadaraClient) -> Result<(), anyhow::Error> {
+    let rpc = madara.get_starknet_client().await;
 
     // deploy account
-    let oz_factory =
-        build_oz_account_factory(rpc, "0x123", FieldElement::from_hex_be(CAIRO_1_ACCOUNT_CONTRACT_CLASS_HASH).unwrap())
-            .await;
+    let oz_factory = build_oz_account_factory(
+        &rpc,
+        "0x789",
+        FieldElement::from_hex_be(CAIRO_1_ACCOUNT_CONTRACT_CLASS_HASH).unwrap(),
+    )
+    .await;
     let account_deploy_txn = build_deploy_account_tx(&oz_factory, FieldElement::ONE);
     let account_address = account_deploy_txn.address();
 
-    let funding_account = create_account(rpc, SIGNER_PRIVATE, ARGENT_CONTRACT_ADDRESS, true);
-
-    let mut txs = madara
-        .create_block_with_txs(vec![
-            Transaction::Execution(funding_account.transfer_tokens(
-                account_address,
-                FieldElement::from_hex_be("0xFFFFFFFFFFFF").unwrap(),
-                None,
-            )),
-            Transaction::AccountDeployment(account_deploy_txn),
-        ])
-        .await?;
+    let funding_account = create_account(&rpc, SIGNER_PRIVATE, ARGENT_CONTRACT_ADDRESS, true);
+
+    let (mut txs, block_number) = {
+        let mut madara_write_lock = madara.write().await;
+        let txs = madara_write_lock
+            .create_block_with_txs(vec![
+                Transaction::Execution(funding_account.transfer_tokens(
+                    account_address,
+                    FieldElement::from_hex_be(MAX_FEE_OVERRIDE).unwrap(),
+                    None,
+                )),
+                Transaction::AccountDeployment(account_deploy_txn),
+            ])
+            .await?;
+        let block_number = rpc.block_number().await?;
+
+        (txs, block_number)
+    };
 
     assert_eq!(txs.len(), 2);
     let account_deploy_tx_result = txs.remove(1);
     match account_deploy_tx_result {
         // passes the validation stage
         Ok(TransactionResult::AccountDeployment(DeployAccountTransactionResult {
-            transaction_hash,
+            transaction_hash: _,
             contract_address,
         })) => {
-            assert_eq!(
```

Comment:
We cannot know the tx_hash beforehand now. Because the nonce will be non deterministic.
So we have no way to check it there.

We need unitest on the compute_tx_hash methods

Reply:
oh ok, makes sense